vac/raw/eth-mls-offchain.md

```diff
@@ -0,0 +1,225 @@
+generates the necessary key material for the next epoch, including the appropriate welcome messages
+for new joiners and new entropy for removed members. In this RFC, the committers only MUST be stewards.
+
+### de-MLS Objects
+
+`Voting proposal message:` Voting proposal messages are different then MLS proposals as voting proposals are basically
+the application message in MLS group. Therefore, the steward can collect the proposals without halting the protocol.
+
+## Flow
+
+General flow is as follows:
+
+- Each time a single `steward` initializes a group and creates and emits a group anouncement (GA) periodically.
+- Meanwhile, each`node`creates and sends their`credential` includes `keyPackage`.
+- Each `member`creates `voting proposals` sends them to from MLS group during epoch E.
+- Meanwhile, the `steward` collects finalized `voting proposals` from MLS group and converts them into
+`MLS proposals`  then sends them with correspondng `commit messages`
+- Evantually, with the commit messages, all members starts the next epoch E+1.
+
+## Creating Voting Proposal
+
+A `member` MAY initializes the voting with the proposal payload
+which is implemented using [protocol buffers v3](https://protobuf.dev/) as follows:
+
+```bash
+syntax = "proto3";
+
+message Proposal {
+string name = 10;                 // Proposal name
+string payload = 11;              // Describes the what is voting fore 
+int32 proposal_id = 12;           // Unique identifier of the proposal
+bytes proposal_owner = 13;        // Public key of the creator
+repeated Vote votes = 14;         // Vote list in the proposal
+int32 expected_voters_count = 15; // Maximum number of distinct voters
+int32 round = 16;                 // Number of Votes
+int64 timestamp = 17;             // Creation time of proposal
+int64 expiration_time = 18;       // Time interval that the proposal is active
+bool liveness_criteria_yes = 19;  // Shows how managing the silent peers vote
+}
+```
+
+```bash
+message Vote {
+int32 vote_id = 20;             // Unique identifier of the vote
+bytes vote_owner = 21;          // Voter's public key
+int64 timestamp = 22;           // Time when the vote was cast
+bool vote = 23;                 // Vote bool value (true/false)
+bytes parent_hash = 24;         // Hash of previous owner's Vote
+bytes received_hash = 25;       // Hash of previous received Vote
+bytes vote_hash = 26;           // Hash of all previously defined fields in Vote
+bytes signature = 27;           // Signature of vote_hash
+}
+```
+
+After the member creates the voting proposal that may include adding or removing a node or member,
+the voting proposal is emitted the network via the MLS application message with a lightweight,
+epoch based voting such as [hashgraphlike consensus.](https://github.com/vacp2p/rfc-index/blob/consensus-hashgraph-like/vac/raw/consensus-hashgraphlike.md)

```

Comment:
not really correct 

in fact we shared message using group app topic and some separate service handle this concrete voting/proposal messages. So we can't say that we share message by voting - we share by p2p network and them use consesus service to get a result or vote